src/dag-builder/index.js

```diff
@@ -33,10 +33,18 @@ async function * dagBuilder (source, ipld, options) {
       const chunker = createChunker(options.chunker, validateChunks(source), options.chunkerOptions)
 
       // item is a file
-      yield fileBuilder(entry, chunker, ipld, options)
+      yield fileBuilder(entry, chunker, ipld, {
+        ...options,
+        mode: entry.mode,
+        mtime: entry.mtime
```

Comment:
Why are these passed as options? - The entry is passed also.

Reply:
Fixed now